cmd/whitesourceExecuteScan.go

```diff
@@ -0,0 +1,370 @@
+			projectsScanned = extractProjectTokensFromStdout(os.Stdout, config, sys)
+		}
+		break
+	}
+	return projectsScanned
+}
+
+// deal with multimodule gradle projects... there's probably a better way of doing this...
+// Problem: Find all project tokens scanned that are apart of multimodule scan.
+// Issue: Only have access to a single project token to begin with (config.ProjectToken)
+func extractProjectTokensFromStdout(stdout *os.File, config whitesourceExecuteScanOptions, sys whitesource.System) []whitesource.Project {
+
+	log.Entry().Info("Running grep command on whitesource stdout...")
+	cmd := exec.Command("grep", "URL: ")
+	projectsInfoBuffer := &bytes.Buffer{}
+	cmd.Stdout = projectsInfoBuffer
+	cmd.Stdin = stdout
+	err := cmd.Run()
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Failed to run 'grep URL: ', exit..")
+	}
+
+	log.Entry().Info("Extracting project tokens from whitesource scan stdout...")
+	ids := []int64{}
+	projectMetaStr := projectsInfoBuffer.String()
+	projectMetas := strings.Split(projectMetaStr, "id=")
+	for _, idStr := range projectMetas {
+		idStr = strings.Split(idStr, "\n")[0]
+		if !strings.HasPrefix(idStr, "[INFO]") {
+			idInt := int64(Atoi(idStr))
+			ids = append(ids, idInt)
+		}
+	}
+
+	projects, err := sys.GetProjectsByIds(config.ProductToken, ids)
+	if err != nil {
+		log.Entry().WithError(err).Errorf("Could not get project by IDs: %s", ids)
+	}
+
+	return projects
+}
+
+// executeNpmScan:
+// generates a configuration file whitesource.config.json with appropriate values from config,
+// installs whitesource yarn plugin and executes the scan
+func executeNpmScan(config whitesourceExecuteScanOptions, command *command.Command) {
+	npmConfig := []byte(fmt.Sprintf(`{
+		"apiKey": "%s",
+		"userKey": "%s",
+		"checkPolicies": true,
+		"productName": "%s",
+		"projectName": "%s",
+		"productVer": "%s",
+		"devDep": true
+	}`, config.OrgToken, config.UserToken, config.ProductName, config.ProjectName, config.ProductVersion))
+
+	writeToFile("./whitesource.config.json", npmConfig, 0644)
+
+	err := command.RunExecutable("yarn", "global", "add", "whitesource")
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Failed to run 'yarn global add whitesource'")
+	}
+	err = command.RunExecutable("yarn", "install")
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Failed to run 'yarn install'")
+	}
+	err = command.RunExecutable("whitesource", "yarn")
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Failed to run 'whitesource yarn'")
+	}
+}
+
+// translated from Groovy DSL: https://github.com/SAP/jenkins-library/blob/4c97231ff94d9f9fbeb1b58c3625c84b434a7de6/vars/whitesourceExecuteScan.groovy#L443
+func resolveProjectIdentifiers(config *whitesourceExecuteScanOptions, command *command.Command, sys whitesource.System) {
+	if config.ProjectName == "" || config.ProductVersion == "" {
+		var gav map[string]string
+
+		switch config.ScanType {
+		case "npm":
+			gav = getNpmGAV()
```

Comment:
in order to get the GAV we recently added functionality in the versioning package.
Here the example for Maven:
https://github.com/SAP/jenkins-library/blob/master/pkg/versioning/maven.go#L55

We could add the same for gradle, ... in this package.

Reply:
Let me know if the latest commit resolves this.

Reply:
not sure if I overlook s.th.
I thought about adding the functionality to `pkg/versioning` similar to https://github.com/SAP/jenkins-library/blob/master/pkg/versioning/maven.go#L55

and then use it like
https://github.com/SAP/jenkins-library/blob/master/cmd/fortifyExecuteScan.go#L64
and
https://github.com/SAP/jenkins-library/blob/master/cmd/fortifyExecuteScan.go#L73

we could leave this for a future optimization though as well since this anyway should be a kind of `alpha` which we should optimize on the go ...

Reply:
> not sure if I overlook s.th.
> I thought about adding the functionality to `pkg/versioning` similar to https://github.com/SAP/jenkins-library/blob/master/pkg/versioning/maven.go#L55
> 
> and then use it like
> https://github.com/SAP/jenkins-library/blob/master/cmd/fortifyExecuteScan.go#L64
> and
> https://github.com/SAP/jenkins-library/blob/master/cmd/fortifyExecuteScan.go#L73
> 
> we could leave this for a future optimization though as well since this anyway should be a kind of `alpha` which we should optimize on the go ...

I went ahead and added it anyways :P
https://github.com/jaylevin/jenkins-library/blob/4c6496679b972deeb59e9b40c816981760ea0c9a/cmd/whitesourceExecuteScan.go#L73